hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -57,6 +57,20 @@
     except ImportError:
         _AnnotatedAlias = ()
 
+TypeAliasTypes: tuple = ()
+try:
+    from typing import TypeAlias as TypingTypeAlias  # type: ignore
+except ImportError:
+    pass  # Is missing for `python<3.10`
+else:  # pragma: no cover
+    TypeAliasTypes += (TypingTypeAlias,)
+try:
+    from typing_extensions import TypeAlias as ExtensionsTypeAlias
+except ImportError:
+    pass  # Is missing for `typing_extensions<3.10`
+else:  # pragma: no cover
+    TypeAliasTypes += (ExtensionsTypeAlias,)
```

Comment:
```suggestion
try:
    TypeAliasTypes += (typing_extensions.TypeAlias,)
except AttributeError:
    pass  # typing_extensions package not available, or older than v3.10
```

And similar for `typing`, which we always import (but might raise `AttributeError`)